src/components/PopoverWithoutOverlay/index.js

```diff
@@ -30,8 +32,11 @@ function Popover(props) {
                 ref: props.withoutOverlayRef,
                 close: props.onClose,
                 anchorRef: props.anchorRef,
-                onCloseCallback: () => Modal.setCloseModal(null),
-                onOpenCallback: () => Modal.setCloseModal(() => props.onClose(props.anchorRef)),
+                onCloseCallback: removeOnClose.current,
+                onOpenCallback: () => {
+                    removeOnClose.current();
```

Comment:
Is this needed? Is this workaround associated with the logic for opening and closing modals?